cpp/open3d/t/geometry/TriangleMesh.cpp

```diff
@@ -1061,6 +1061,188 @@ TriangleMesh TriangleMesh::SelectFacesByMask(const core::Tensor &mask) const {
     return result;
 }
 
+///
+/// This function shrinks a vertex attribute tensor to have length new_size.
+/// This function is used in RemoveDuplicateVerticesWorker.
+/// Assumptions: 
+/// 1. The new_size is the number of unique vertices in the mesh.
+/// 2. The attribute tensor has been updated during duplicate removal in such a way that
+///    attributes corresponding to unique vertices are moved to the beginning and the rest
+///    can be discarded.
+/// \param attrib The attribute whose tensor has to be shrunk.
+/// \param mesh The mesh from which duplicates have to be removed.
+/// \param new_size The size to shrink the attribute tensor to.
+void ShrinkVertexAttributeTensor(const std::string &attrib, TriangleMesh &mesh, int64_t new_size)    
+{
+    auto old_tensor = mesh.GetVertexAttr(attrib);
+    std::vector<decltype(new_size)> old_shape_vec(old_tensor.GetShape());
+    assert (new_size <= old_shape_vec[0]);
```

Comment:
Use `utility::LogError` instead of `assert` for runtime checks to allow runtime user control over checks.
For debug only checks use OPEN3D_ASSERT macro